src/server/gc_worker/applied_lock_collector.rs

```diff
@@ -407,6 +417,15 @@ impl AppliedLockCollector {
     /// Starts collecting applied locks whose `start_ts` <= `max_ts`. Only one `max_ts` is valid
     /// at one time.
     pub fn start_collecting(&self, max_ts: TimeStamp, callback: Callback<()>) -> Result<()> {
+        // Before starting collecting, check the concurrency manager to avoid later prewrite
+        // requests uses a min_commit_ts less than the safepoint.
+        // `max_ts` here is the safepoint of the current round of GC.
+        self.concurrency_manager.update_max_ts(max_ts);
+        self.concurrency_manager
+            .read_range_check(None, None, |k, l| {
+                Lock::check_ts_conflict(Cow::Borrowed(l), k, max_ts, &TsSet::Empty)
+                    .map_err(MvccError::from)
+            })?;
```

Comment:
Is there a test for this?

Reply:
For the current implementation of TiDB, GC will simply fail.